controllers/flinkcluster_reconciler.go

```diff
@@ -403,20 +403,34 @@ func (reconciler *ClusterReconciler) reconcileJob() (ctrl.Result, error) {
 
 	// Delete
 	if desiredJob == nil && observedJob != nil {
-		if len(jobID) > 0 {
+		// Cancel Flink job if it is live
+		// 1) In the case of which savepoint was triggered, after it is completed, proceed to delete step.
+		// 2) When savepoint was skipped, continue to next step immediately.
+		//
+		// If savepoint or cancellation was failed, the control state will fall to the failed in the updater.
+		if len(jobID) > 0 && len(observed.flinkRunningJobIDs) == 1 {
```

Comment:
Why do we need to add `len(observed.flinkRunningJobIDs) == 1`?

Reply:
https://github.com/elanv/flink-on-k8s-operator/blob/6538a03434fa7f3a2082b820be65e2dd4e419837/controllers/flinkcluster_observer.go#L250-L269

Like 264 line, in rare cases, it could be the ID of a stopped job, so I added the condition.